sdk/iot/hub/tests/cmocka/test_az_iot_hub_client_c2d.c

```diff
@@ -204,29 +189,66 @@ static void test_az_iot_hub_client_c2d_parse_received_topic_URL_ENCODED_succeed(
 
   az_iot_hub_client_c2d_request out_request;
 
-  assert_return_code(
+  assert_int_equal(
       az_iot_hub_client_c2d_parse_received_topic(&client, received_topic, &out_request), AZ_OK);
 
   az_pair pair;
-  assert_return_code(az_iot_hub_client_properties_next(&out_request.properties, &pair), AZ_OK);
+  assert_int_equal(az_iot_hub_client_properties_next(&out_request.properties, &pair), AZ_OK);
   assert_true(az_span_is_content_equal(pair.key, AZ_SPAN_FROM_STR("%24.to")));
   assert_true(az_span_is_content_equal(pair.value,
   AZ_SPAN_FROM_STR("%2Fdevices%2Fuseragent_c%2Fmessages%2FdeviceBound")));
   
-  assert_return_code(az_iot_hub_client_properties_next(&out_request.properties, &pair), AZ_OK);
+  assert_int_equal(az_iot_hub_client_properties_next(&out_request.properties, &pair), AZ_OK);
   assert_true(az_span_is_content_equal(pair.key, AZ_SPAN_FROM_STR("abc")));
   assert_true(az_span_is_content_equal(pair.value, AZ_SPAN_FROM_STR("123")));
 
-  assert_return_code(az_iot_hub_client_properties_next(&out_request.properties, &pair), AZ_OK);
+  assert_int_equal(az_iot_hub_client_properties_next(&out_request.properties, &pair), AZ_OK);
   assert_true(az_span_is_content_equal(pair.key, AZ_SPAN_FROM_STR("ghi")));
   assert_true(az_span_is_content_equal(pair.value, AZ_SPAN_FROM_STR("%2Fsome%2Fthing")));
 
-  assert_return_code(az_iot_hub_client_properties_next(&out_request.properties, &pair), AZ_OK);
+  assert_int_equal(az_iot_hub_client_properties_next(&out_request.properties, &pair), AZ_OK);
   assert_true(az_span_is_content_equal(pair.key, AZ_SPAN_FROM_STR("jkl")));
   assert_true(az_span_is_content_equal(pair.value,
   AZ_SPAN_FROM_STR("%2Fsome%2Fthing%2F%3Fbla%3Dbla")));
 }
 
+static void test_az_iot_hub_client_c2d_parse_received_topic_fail(void** state)
+{
+  (void)state;
+
+  az_iot_hub_client client;
+  az_iot_hub_client_options options = az_iot_hub_client_options_default();
+  assert_true(
+      az_iot_hub_client_init(&client, test_device_hostname, test_device_id, &options) == AZ_OK);
+
+  az_span received_topic = test_parse_method_topic_fail;
+
+  az_iot_hub_client_c2d_request out_request;
+
+  assert_int_equal(
+      az_iot_hub_client_c2d_parse_received_topic(
```

Comment:
Nit: assert(expected, actual) is easier to read since the first arg is usually short.

Reply:
Understood. That could come in a later PR since I think all of hub is done this way.